pallets/mt/src/lib.rs

```diff
@@ -215,22 +215,13 @@ pub mod pallet {
 		fn on_initialize(_n: T::BlockNumber) -> Weight {
 			let mut temp_hashes: Vec<T::Element> = Vec::with_capacity(T::MaxTreeDepth::get() as usize);
 			let default_zero = T::DefaultZeroElement::get();
-			let default_zero_bytes = default_zero.to_bytes();
-			let mut temp_hash = T::Hasher::hash_two(
-				default_zero_bytes.clone(),
-				default_zero_bytes.clone()
-			).unwrap_or(default_zero_bytes.to_vec());
-			// add temp hash to collection
-			temp_hashes.push(T::Element::from_vec(temp_hash.clone()));
-
-			for _ in 1..=T::MaxTreeDepth::get() {
-				temp_hash = T::Hasher::hash_two(
-					&temp_hash,
-					&temp_hash
-				).unwrap_or(default_zero_bytes.to_vec());
+			temp_hashes.push(default_zero);
+			let mut temp_hash = default_zero.to_bytes().to_vec();
+			for _ in 0..T::MaxTreeDepth::get() {
+				temp_hash = T::Hasher::hash_two(&temp_hash, &temp_hash).unwrap();
```

Comment:
One problem with setting zeroes up on_init now that I think about it is ensuring the parameters are there (or this will throw). Since we'll likely jumpstart this on genesis it should be fine (we will put params in genesis storage).  I'm curious if you have thought about this at all?

Reply:
Yes, I do agree, in this case, it will just crash (due to the `unwrap` here).
In tests, I do init the hasher pallet. I guess we need a way to check this first, instead of just falling.

The other idea about adding the zero hashes in the genesis sounds good, also we have a way to send them during runtime too. so I guess it is not a big deal, for now.
also, we could remove this hook as it does not make sense now.

Reply:
I think it's useful for testing and potentially on runtime upgrades when new pallets are initialized (assuming a new hasher has parameters). It is a good first step to covering the non-genesis flow.